modules/core/PrimDisc.cpp

```diff
@@ -38,6 +40,9 @@ namespace rt {
 		Vec3f r = m_radius * normalize(Vec3f::all(1));
 		r = CTransform::vector(r, T);
 		m_radius = static_cast<float>(norm(r));
+          
+          // Accumulate transformation in the transformation matrix
+		m_t = m_t * T;
```

Comment:
I think we need to move this also to CPrim. So CPrim should have its own implementation of the transform method